subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,1930 @@
+    }
+
+    def "can add dependency with actions on suite using a #desc"() {
+        given:
+        buildFile << """
+        plugins {
+          id 'java-library'
+        }
+
+        ${mavenCentralRepository()}
+
+        testing {
+            suites {
+                test {
+                    dependencies {
+                        implementation($dependencyNotation) {
+                            exclude group: '$collectionsGroup', module: '$collectionsName'
+                        }
+                    }
+                }
+            }
+        }
+
+        tasks.register('checkConfiguration') {
+            dependsOn test
+            doLast {
+                def testCompileClasspathFileNames = configurations.testCompileClasspath.files*.name
+                assert testCompileClasspathFileNames.containsAll('${beanUtilsName}-${beanUtilsVer}.jar')
+                assert !testCompileClasspathFileNames.contains('${collectionsName}-${collectionsVer}.jar'): 'excluded dependency'
+            }
+        }
+        """
+
+        file('src/main/org/sample/Person.java') << """
+            package org.sample;
+
+            public class Person {
+                private String name;
+                private int age;
+
+                public String getName() {
+                    return name;
+                }
+
+                public void setName(String name) {
+                    this.name = name;
+                }
+
+                public int getAge() {
+                    return age;
+                }
+
+                public void setAge(int age) {
+                    this.age = age;
+                }
+            }
+        """
+
+        file('src/test/org/samplePersonTest.java') << """
+            package org.sample;
+
+            import org.apache.commons.beanutils.PropertyUtils;
+
+            public class PersonTest {
+                @Test
+                public void testPerson() {
+                    Object person = new Person();
+                    PropertyUtils.setSimpleProperty(person, "name", "Bart Simpson");
+                    PropertyUtils.setSimpleProperty(person, "age", 38);
+                    assertEquals("Bart Simpson", person.getName());
+                    assertEquals(38, person.getAge());
+                }
+            }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        desc                | dependencyNotation
+        'GAV string'        | gavStr(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'GAV map'           | gavMap(beanUtilsGroup, beanUtilsName, beanUtilsVer)
+        'named args'        | namedArgs(beanUtilsGroup, beanUtilsName, beanUtilsVer)
```

Comment:
I think this one is similar to the other test with a lot of indirection, but it's not quite as bad (since you only have one dependency). I would just put the exact string here as the notation vs building it from each of those methods.

Reply:
I broke it up anyway, to be consistent with the other.